catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/OverrideAttributesSupportTest.java

```diff
@@ -154,28 +157,35 @@ public void testOverrideMetacardIgnoreType() throws URISyntaxException {
   }
 
   @Test
-  public void testOverrideMetacard() throws URISyntaxException {
+  public void testOverrideMetacardDescriptors() throws URISyntaxException {
     MetacardImpl metacard = new MetacardImpl();
     metacard.setMetadata("original");
     metacard.setTitle("original");
     metacard.setId("original");
     metacard.setResourceURI(new URI("content:stuff"));
     MetacardImpl overrideMetacard =
         new MetacardImpl(
-            new MetacardTypeImpl("other", metacard.getMetacardType().getAttributeDescriptors()));
+            new MetacardTypeImpl(
+                "other",
+                metacard.getMetacardType(),
+                new MediaAttributes().getAttributeDescriptors()));
     overrideMetacard.setTitle("updated");
     overrideMetacard.setId("updated");
     overrideMetacard.setMetadata("updated");
     overrideMetacard.setResourceURI(new URI("content:newstuff"));
+    overrideMetacard.setAttribute(Media.DURATION, 1.0);
 
     Metacard updatedMetacard =
-        OverrideAttributesSupport.overrideMetacard(metacard, overrideMetacard, false, false);
+        OverrideAttributesSupport.overrideMetacard(metacard, overrideMetacard, false);
 
     assertThat(updatedMetacard.getMetadata(), is("updated"));
     assertThat(updatedMetacard.getTitle(), is("updated"));
     assertThat(updatedMetacard.getResourceURI().toString(), is("content:newstuff"));
     assertThat(updatedMetacard.getId(), is("original"));
-    assertThat(updatedMetacard.getMetacardType().getName(), is("other"));
+    assertThat(updatedMetacard.getMetacardType().getName(), is("ddf.metacard"));
+    assertThat(
+        updatedMetacard.getMetacardType().getAttributeDescriptor(Media.DURATION), notNullValue());
+    assertThat(updatedMetacard.getAttribute(Media.DURATION).getValue(), is(1.0));
   }
 
   @Test
```

Comment:
❗️We should also add tests for the case where `onlyFillNull` is `true`.

Reply:
Added test for this case